src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -59,75 +70,210 @@ const getLambdaHandlerCandidates = async ({ uri }: { uri: vscode.Uri }): Promise
+        let manifestPath: string | undefined
         if (args.isDebug) {
             debugPort = await getDebugPort()
+            const {debugHandlerName} = await makeLambdaDebugFile({
+                handlerName: args.handlerName,
+                debugPort: debugPort,
+                outputDir: samProjectCodeRoot,
+            })
+            handlerName = debugHandlerName
+            manifestPath = await makePythonDebugManifest({
+                samProjectCodeRoot,
+                outputDir: baseBuildDir
+            })
+        } else {
+            handlerName = args.handlerName
         }
-        // TODO: Figure out Python specific params and create PythonDebugConfiguration if needed
-        const debugConfig: DebugConfiguration = {
-            type: PYTHON_LANGUAGE,
-            request: 'attach',
-            name: 'SamLocalDebug',
-            preLaunchTask: undefined,
-            address: 'localhost',
-            port: debugPort!,
-            localRoot: samProjectCodeRoot,
-            remoteRoot: '/var/task',
-            skipFiles: []
-        }
+        const inputTemplatePath = await makeInputTemplate({
+            baseBuildDir,
+            codeDir: samProjectCodeRoot,
+            documentUri: args.document.uri,
+            handlerName,
+            runtime,
+            workspaceUri: args.workspaceFolder.uri
+        })
+        logger.info(`pythonCodeLensProvider.initialize: ${
+            JSON.stringify({samProjectCodeRoot, inputTemplatePath, handlerName, manifestPath}, undefined, 2)
+        }`)
+
+        const channelLogger = getChannelLogger(toolkitOutputChannel)
+        const codeDir = samProjectCodeRoot
+        const samTemplatePath: string = await executeSamBuild({
+            baseBuildDir,
+            channelLogger,
+            codeDir,
+            inputTemplatePath,
+            manifestPath,
+            samProcessInvoker: processInvoker,
 
-        const localLambdaRunner: LocalLambdaRunner = new LocalLambdaRunner(
+        })
+    
+        const debugConfig: PythonDebugConfiguration = makeDebugConfig({debugPort, samProjectCodeRoot})
+        await invokeLambdaFunction({
+            baseBuildDir,
+            channelLogger,
             configuration,
-            args,
-            debugPort,
-            runtime,
-            toolkitOutputChannel,
-            processInvoker,
-            taskInvoker,
             debugConfig,
-            samProjectCodeRoot,
-            // TODO: Use onWillAttachDebugger &/or onDidSamBuild to enable debugging support
-        )
-
-        await localLambdaRunner.run()
+            samTaskInvoker: taskInvoker,
+            samTemplatePath,
+            documentUri: args.document.uri,
+            handlerName,
+            isDebug: args.isDebug,
+            onWillAttachDebugger: async () => {
+                // TODO: Find out why debugger can't detach without introducing delay
+                if (process.platform === 'darwin') {
```

Comment:
Were you able to test this on other OSes?

Reply:
Works on Windows... I'd guess the safest logic would be to test for os != windows but I don't want to introduce an un-needed delay. Would be great to have a linux tester.

Reply:
Agreed, if we publish with this workaround, we need to validate on linux.

Reply:
@bryceitoc9 will do a linux test. 

I recommend that we prepare this PR as a windows-only change for now, by
* disabling all of the Python codelenses for non-windows OSes
* remove the hack sleep block

This PR does the heavy lifting towards supporting Python, and gives us a great jumping off point for solving the timing issues on Mac. If we find out Linux has a similar issue, we can easily enable the CodeLenses in another change.

Thoughts?

Reply:
That's reasonable. Is there a good reason not to enable the `run` code lens for Python?

Reply:
My mistake, I was conflating a separate issue (completely not related to this PR) related to system pathing. Once we know more about that, we can decide separate from this PR how we will handle run/configure codelenses.

Let's leave run and configure for now, and only show Debug on Windows

Reply:
Done.
✔️